src/Illuminate/Console/Process/PendingProcess.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Illuminate\Console\Process;
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Traits\Macroable;
+use Symfony\Component\Process\Process;
+
+class PendingProcess
+{
+    use Macroable;
+
+    /**
+     * The process's arguments.
+     *
+     * @var iterable<array-key, string>
+     */
+    protected $arguments;
+
+    /**
+     * The process's path.
+     *
+     * @var string|null
+     */
+    protected $path;
+
+    /**
+     * Create a new Pending Process instance.
+     *
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        // ..
+    }
+
+    /**
+     * Sets the process's arguments.
+     *
+     * @param  iterable<array-key, string>  $arguments
+     * @return $this
+     */
+    public function withArguments($arguments)
+    {
+        return tap($this, fn () => $this->arguments = array_merge($this->arguments ?? [], array_values($arguments)));
+    }
+
+    /**
+     * Sets the process's path.
+     *
+     * @param  string  $path
+     * @return $this
+     */
+    public function path($path)
+    {
+        return tap($this, fn () => $this->path = $path);
```

Comment:
I do use `tap(...)` a lot on my projects, and the usage in `run()` method is something I think `tap()` brings simplicity.

But here and in the `withArgument` method, I don't see the advantage of using it. No temporary variables would be needed, both are simple assignments, and I know discussing micro-optimization is out of question, but it adds two function calls for no benefit (in my opinion).

Also I think in such cases it is actually more complex to skim through, and I don't know any other *"wither"* or fluent setter in the framework code which use this code style.

Reply:
This PR is a work in progress, ignore the code for now.